lib/reek/cli/silencer.rb

```diff
@@ -3,11 +3,24 @@ module CLI
     # CLI silencer
     # @api private
     module Silencer
-      def self.silently
-        old_verbose, $VERBOSE = $VERBOSE, nil
+      def self.silently(stdout: false, stderr: false)
+        old_verbose = $VERBOSE
+        $VERBOSE = nil
+        if stdout
+          old_stdout = $stdout
+          $stdout = StringIO.new
+        end
+
+        if stderr
+          old_stderr = $stderr
+          $stderr = StringIO.new
+        end
+
         yield if block_given?
       ensure
         $VERBOSE = old_verbose
+        $stderr = old_stderr if stderr
+        $stdout = old_stdout if stdout
```

Comment:
From the ‘made the comment, deleted, made it again, deleted, made it third time’ department: I’m thinking whether reassigning to `STDERR` and `STDOUT` wouldn’t be more canonical (and significantly simpler). They’re supposed to hold the references to the canonical streams, and the only reason for caching `old_stderr`/`old_stdout` by hand would be if this silencer was itself embedded in another piece of code tha alters `$stderr`/`$stdout`.

Also, rather than Boolean flags, how about making the kwargs the target streams (so the caller would pass `StringIO.new` or whatever it needs rather than `true`)? And how about making this a `module_function`? :)

```ruby
    module Silencer
      module_function

      def silently(stdout: nil, stderr: nil)
        old_verbose = $VERBOSE
        $VERBOSE = false
        $stdout = stdout if stdout
        $stderr = stderr if stderr
        yield
      ensure
        $VERBOSE = old_verbose
        $stderr = STDERR
        $stdout = STDOUT
      end
    end
```